packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1530,6 +1577,16 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
         return summarizeResult as IChannelSummarizeResult;
     }
 
+    private shouldRunGC(): boolean {
+        // Can override with localStorage flag.
+        return localStorageEnableGC() ?? (
```

Comment:
I don't like this as a member function of this class, and i don't like the value changing at runtime, we should use this data to build the initial configuration, and then only depend on the config, two sources of truth like this always leads to bugs

Reply:
I agree with you.
I changed to readonly properties: `shouldRunGC` and `disableIsolatedChannels`. Now I no longer modify the passed in `runtimeOptions` for those.

I kept the `shouldWriteMetadata` getter, because it is a computed property using all readonly properties.